diff -ur qt-x11-free-3.0.5.orig/configure qt-x11-free-3.0.5/configure
--- qt-x11-free-3.0.5.orig/configure	Thu Jul  4 08:37:30 2002
+++ qt-x11-free-3.0.5/configure	Wed Jul 31 12:53:08 2002
@@ -2297,8 +2297,8 @@
 #define QT_BUILD_KEY "$QT_BUILD_KEY"
 EOF
 [ '!' -z "$LicenseKey" ] && echo "#define QT_PRODUCT_LICENSEKEY \"$LicenseKey\"" >>$outpath/include/qconfig.h.new
-[ '!' -z "$QT_INSTALL_PLUGINS" ] && echo "#define QT_INSTALL_PLUGINS ${QT_INSTALL_PLUGINS}" >>$outpath/include/qconfig.h.new
-[ '!' -z "$QT_INSTALL_DATA" ]    && echo "#define QT_INSTALL_DATA    ${QT_INSTALL_DATA}"    >>$outpath/include/qconfig.h.new
+[ '!' -z "$QT_INSTALL_PLUGINS" ] && echo "#define QT_INSTALL_PLUGINS \"${QT_INSTALL_PLUGINS}\"" >>$outpath/include/qconfig.h.new
+[ '!' -z "$QT_INSTALL_DATA" ]    && echo "#define QT_INSTALL_DATA    \"${QT_INSTALL_DATA}\""    >>$outpath/include/qconfig.h.new
 
 # avoid unecessary rebuilds by copying only if qconfig.h has changed
 if cmp -s $outpath/include/qconfig.h $outpath/include/qconfig.h.new; then
Only in qt-x11-free-3.0.5.orig/include: qt_windows.h
