diff -urN qt-x11-free-3.0.4.orig/configure qt-x11-free-3.0.4/configure
--- qt-x11-free-3.0.4.orig/configure	Mon Apr 22 22:03:46 2002
+++ qt-x11-free-3.0.4/configure	Sun May  5 21:42:15 2002
@@ -1920,8 +1920,8 @@
 #define QT_PRODUCT_LICENSE "$Products"
 EOF
 [ '!' -z "$LicenseKey" ] && echo "#define QT_PRODUCT_LICENSEKEY \"$LicenseKey\"" >>qconfig.h.new
-[ '!' -z "$QT_INSTALL_PLUGINS" ] && echo "#define QT_INSTALL_PLUGINS ${QT_INSTALL_PLUGINS}" >>qconfig.h.new
-[ '!' -z "$QT_INSTALL_DATA" ]    && echo "#define QT_INSTALL_DATA    ${QT_INSTALL_DATA}"    >>qconfig.h.new
+[ '!' -z "$QT_INSTALL_PLUGINS" ] && echo "#define QT_INSTALL_PLUGINS ${QT_INSTALL_PLUGINS}" >>$outpath/include/qconfig.h.new
+[ '!' -z "$QT_INSTALL_DATA" ]    && echo "#define QT_INSTALL_DATA    ${QT_INSTALL_DATA}"    >>$outpath/include/qconfig.h.new
 
 # avoid unecessary rebuilds by copying only if qconfig.h has changed
 if cmp -s $outpath/include/qconfig.h $outpath/include/qconfig.h.new; then
