--- qt-x11-free-3.1.1/src/kernel/qcursor.cpp.orig	Mon Dec  9 10:40:46 2002
+++ qt-x11-free-3.1.1/src/kernel/qcursor.cpp	Wed Feb 19 15:53:59 2003
@@ -102,7 +102,7 @@
     \value PointingHandCursor - a pointing hand
     \value ForbiddenCursor - a slashed circle
     \value WhatsThisCursor - an arrow with a question mark
-    \value BitmapCursor
+    \value WorkingCursor - an arrow with a clock
 
     ArrowCursor is the default for widgets in a normal state.
 
--- qt-copy/src/kernel/qcursor_x11.cpp~	2003-05-16 23:14:40.000000000 +0200
+++ qt-copy/src/kernel/qcursor_x11.cpp	2003-05-17 02:27:09.000000000 +0200
@@ -51,6 +51,8 @@
 // Define QT_USE_APPROXIMATE_CURSORS when compiling if you REALLY want to
 // use the ugly X11 cursors.
 
+#define QT_USE_APPROXIMATE_CURSORS
+
 /*****************************************************************************
   Internal QCursorData class
  *****************************************************************************/
@@ -118,6 +120,7 @@
 QT_STATIC_CONST_IMPL QCursor & Qt::pointingHandCursor = cursorTable[13];
 QT_STATIC_CONST_IMPL QCursor & Qt::forbiddenCursor = cursorTable[14];
 QT_STATIC_CONST_IMPL QCursor & Qt::whatsThisCursor = cursorTable[15];
+QT_STATIC_CONST_IMPL QCursor & Qt::workingCursor = cursorTable[16];
 
 
 QCursor *QCursor::find_cur( int shape )		// find predefined cursor
@@ -723,6 +726,9 @@
     case WaitCursor:
 	sh = XC_watch;
 	break;
+    case WorkingCursor:
+	sh = XC_clock;
+	break;
     case IbeamCursor:
 	sh = XC_xterm;
 	break;
--- qt-x11-free-3.1.1/src/kernel/qnamespace.h.orig	Mon Dec  9 10:40:45 2002
+++ qt-x11-free-3.1.1/src/kernel/qnamespace.h	Wed Feb 19 15:53:59 2003
@@ -722,7 +722,8 @@
 	PointingHandCursor,
 	ForbiddenCursor,
 	WhatsThisCursor,
-	LastCursor	= WhatsThisCursor,
+	WorkingCursor,
+	LastCursor	= WorkingCursor,
 	BitmapCursor	= 24
     };
 
@@ -746,6 +747,7 @@
     QT_STATIC_CONST QCursor & pointingHandCursor;	// pointing hand
     QT_STATIC_CONST QCursor & forbiddenCursor;	// forbidden cursor (slashed circle)
     QT_STATIC_CONST QCursor & whatsThisCursor;  // arrow with a question mark
+    QT_STATIC_CONST QCursor & workingCursor;  // arrow with a question mark
 
 
     enum TextFormat {
