--- configs/linux-g++-shared.orig	Fri Jul 27 19:44:50 2001
+++ configs/linux-g++-shared	Fri Jul 27 19:48:59 2001
@@ -32,7 +32,7 @@
 SYSCONF_LIBS_YACC	= 
 
 # Linking applications
-SYSCONF_LINK		= g++
+SYSCONF_LINK		= objprelink $(OBJECTS) $(OBJMOC) && g++
 SYSCONF_LFLAGS		= 
 SYSCONF_LIBS		= 
 
@@ -58,7 +58,8 @@
 #
 SYSCONF_LINK_SHLIB	= g++
 SYSCONF_LINK_TARGET_SHARED	= lib$(TARGET).so.$(VER_MAJ).$(VER_MIN).$(VER_PATCH)
-SYSCONF_LINK_LIB_SHARED	=  $(SYSCONF_LINK_SHLIB) -shared -Wl,-soname,lib$(TARGET).so.$(VER_MAJ) \
+SYSCONF_LINK_LIB_SHARED	=  objprelink $(OBJECTS) $(OBJMOC) && \
+			      $(SYSCONF_LINK_SHLIB) -shared -Wl,-soname,lib$(TARGET).so.$(VER_MAJ) \
 				     $(LFLAGS) -o $(SYSCONF_LINK_TARGET_SHARED) \
 				     $(OBJECTS) $(OBJMOC) $(LIBS) && \
 				 mv $(SYSCONF_LINK_TARGET_SHARED) $(DESTDIR); \
