diff -urN qt-2.3.0.org/src/tools/qtextcodec.cpp qt-2.3.0/src/tools/qtextcodec.cpp
--- qt-2.3.0.org/src/tools/qtextcodec.cpp	Wed Apr  4 15:46:46 2001
+++ qt-2.3.0/src/tools/qtextcodec.cpp	Wed Apr  4 16:56:42 2001
@@ -58,6 +58,7 @@
 #include <stdlib.h>
 #include <ctype.h>
 #include <locale.h>
+#include <langinfo.h>
 
 
 static QList<QTextCodec> * all = 0;
@@ -558,6 +559,19 @@
 #ifdef _OS_WIN32_
     localeMapper = new QWindowsLocalCodec;
 #else
+    
+    // Use nl_langinfo first
+    QCString origlocale = setlocale( LC_CTYPE, NULL );
+    setlocale( LC_CTYPE, "" );
+    char * codeset=nl_langinfo(CODESET); 
+    setlocale( LC_CTYPE, origlocale.data() );
+    if (codeset) {
+	localeMapper = codecForName( codeset );
+	if ( localeMapper )
+	    return localeMapper;
+    }
+	
+    
     // Very poorly defined and followed standards causes lots of code
     // to try to get all the cases...
 
