Index: qdom.cpp
===================================================================
RCS file: /home/kde/qt-copy/src/xml/qdom.cpp,v
retrieving revision 1.40
retrieving revision 1.41
diff -u -r1.40 -r1.41
--- qt-copy/src/xml/qdom.cpp~	23 Jul 2003 18:14:04 -0000	1.40
+++ qt-copy/src/xml/qdom.cpp	13 Aug 2003 12:04:51 -0000	1.41
@@ -1459,14 +1459,14 @@
 
 QDomDocumentPrivate* QDomNodePrivate::ownerDocument()
 {
-    if ( hasParent ) {
-	QDomNodePrivate* p = this;
-	while ( p && !p->isDocument() )
-	    p = p->parent();
-
-	return (QDomDocumentPrivate*)p;
+    QDomNodePrivate* p = this;
+    while ( p && !p->isDocument() ) {
+	if ( !p->hasParent )
+	    return (QDomDocumentPrivate*)p->ownerNode;
+	p = p->parent();
     }
-    return (QDomDocumentPrivate*)ownerNode;
+
+    return (QDomDocumentPrivate*)p;
 }
 
 QDomNodePrivate* QDomNodePrivate::cloneNode( bool deep )
