--- qt-3.0.5/src/codecs/qtextcodec.cpp.old	2002-07-08 13:26:48.000000000 +0300
+++ qt-3.0.5/src/codecs/qtextcodec.cpp	2002-08-08 15:57:08.000000000 +0300
@@ -619,7 +619,7 @@
     "eo", 0 };
 
 static const char * const iso8859_4locales[] = {
-    "ee", "ee_EE", "lv", "lv_LV", 0 };
+    "ee", "ee_EE", 0 };
 
 static const char * const iso8859_5locales[] = {
     "mk", "mk_MK", "sp", "sp_YU", 0 };
@@ -643,7 +643,7 @@
     "tr", "tr_TR", "turkish", 0 };
 
 static const char * const iso8859_13locales[] = {
-    "lt", "lt_LT", 0 };
+    "lt", "lt_LT", "lv", "lv_LV", 0 };
 
 static const char * const iso8859_15locales[] = {
     "fr", "fi", "french", "finnish", "et", "et_EE", 0 };
