--- qt3/qmake/project.cpp~	2011-10-30 19:55:57.000000000 +0100
+++ qt3/qmake/project.cpp	2013-10-31 19:11:18.795183368 +0100
@@ -685,7 +685,7 @@
 //	    fprintf(stderr,"Current QT version number: " + ver + "\n");
 	    if (ver != "" && ver != test_version) {
 		ver = test_version;
-		fprintf(stderr,"Changed QT version number to " + test_version + "!\n");
+		fprintf(stderr,"Changed QT version number to %s!\n", test_version.latin1());
 	    }
 	}
     }
--- qt3/src/tools/qgdict.cpp~	2011-10-30 19:55:58.000000000 +0100
+++ qt3/src/tools/qgdict.cpp	2013-10-31 19:18:57.476803248 +0100
@@ -846,11 +846,11 @@
     QString line;
     line.fill( '-', 60 );
     double real, ideal;
-    qDebug( line.ascii() );
+    qDebug( "%s", line.ascii() );
     qDebug( "DICTIONARY STATISTICS:" );
     if ( count() == 0 ) {
 	qDebug( "Empty!" );
-	qDebug( line.ascii() );
+	qDebug( "%s", line.ascii() );
 	return;
     }
     real = 0.0;
@@ -871,7 +871,7 @@
 	while ( b-- )
 	    *pbuf++ = '*';
 	*pbuf = '\0';
-	qDebug( buf );
+	qDebug( "%s", buf );
 	i++;
     }
     qDebug( "Array size = %d", size() );
@@ -879,7 +879,7 @@
     qDebug( "Real dist  = %g", real );
     qDebug( "Rand dist  = %g", ideal );
     qDebug( "Real/Rand  = %g", real/ideal );
-    qDebug( line.ascii() );
+    qDebug( "%s", line.ascii() );
 #endif // QT_DEBUG
 }
 
--- qt3/src/tools/qgcache.cpp~	2011-10-30 19:55:58.000000000 +0100
+++ qt3/src/tools/qgcache.cpp	2013-10-31 19:21:07.980236714 +0100
@@ -641,7 +641,7 @@
 #if defined(QT_DEBUG)
     QString line;
     line.fill( '*', 80 );
-    qDebug( line.ascii() );
+    qDebug( "%s", line.ascii() );
     qDebug( "CACHE STATISTICS:" );
     qDebug( "cache contains %d item%s, with a total cost of %d",
 	   count(), count() != 1 ? "s" : "", tCost );
@@ -662,7 +662,7 @@
 	   lruList->dumps != 1 ? "have" : "has", lruList->dumpCosts );
     qDebug( "Statistics from internal dictionary class:" );
     dict->statistics();
-    qDebug( line.ascii() );
+    qDebug( "%s", line.ascii() );
 #endif
 }
 
--- qt3/src/moc/moc.y~	2013-11-07 23:01:46.000000000 +0100
+++ qt3/src/moc/moc.y	2013-11-07 23:13:13.792254449 +0100
@@ -2831,11 +2831,6 @@ int generateClassInfos()
 
 void generateClass()		      // generate C++ source code for a class
 {
-    const char *hdr1 = "/****************************************************************************\n"
-		 "** %s meta object code from reading C++ file '%s'\n**\n";
-    const char *hdr2 = "** Created: %s\n"
-    const char *hdr3 = "** WARNING! All changes made in this file will be lost!\n";
-    const char *hdr4 = "*****************************************************************************/\n\n";
     int   i;
 
     if ( skipClass )				// don't generate for class
@@ -2871,10 +2866,11 @@ void generateClass()		      // generate
 	    i--;				// skip path
 	if ( i >= 0 )
 	    fn = &g->fileName[i];
-	fprintf( out, hdr1, (const char*)qualifiedClassName(),(const char*)fn);
-	fprintf( out, hdr2, (const char*)dstr );
-	fprintf( out, hdr3 );
-	fprintf( out, hdr4 );
+	fprintf( out, "/****************************************************************************\n"
+		 "** %s meta object code from reading C++ file '%s'\n**\n", (const char*)qualifiedClassName(),(const char*)fn);
+	fprintf( out, "** Created: %s\n", (const char*)dstr );
+	fprintf( out, "%s", "** WARNING! All changes made in this file will be lost!\n");
+	fprintf( out, "%s", "*****************************************************************************/\n\n");
 
 	if ( !g->noInclude ) {
 	    /*

--- qt3/src/moc/moc_yacc.cpp~	2011-10-30 19:55:58.000000000 +0100
+++ qt3/src/moc/moc_yacc.cpp	2013-11-07 23:34:26.422189728 +0100
@@ -2870,12 +2870,6 @@
 
 void generateClass()		      // generate C++ source code for a class
 {
-    const char *hdr1 = "/****************************************************************************\n"
-		 "** %s meta object code from reading C++ file '%s'\n**\n";
-    const char *hdr2 = "** Created: %s\n"
-		 "**      by: The Qt MOC ($Id: qt/moc_yacc.cpp   3.3.8   edited Feb 2 14:59 $)\n**\n";
-    const char *hdr3 = "** WARNING! All changes made in this file will be lost!\n";
-    const char *hdr4 = "*****************************************************************************/\n\n";
     int   i;
 
     if ( skipClass )				// don't generate for class
@@ -2911,10 +2905,10 @@
 	    i--;				// skip path
 	if ( i >= 0 )
 	    fn = &g->fileName[i];
-	fprintf( out, hdr1, (const char*)qualifiedClassName(),(const char*)fn);
-	fprintf( out, hdr2, (const char*)dstr );
-	fprintf( out, hdr3 );
-	fprintf( out, hdr4 );
+	fprintf( out, "/****************************************************************************\n** %s meta object code from reading C++ file '%s'\n**\n", (const char*)qualifiedClassName(),(const char*)fn);
+	fprintf( out, "** Created: %s\n**      by: The Qt MOC ($Id: qt/moc_yacc.cpp   3.3.8   edited Feb 2 14:59 $)\n**\n", (const char*)dstr );
+	fprintf( out, "** WARNING! All changes made in this file will be lost!\n");
+	fprintf( out, "*****************************************************************************/\n\n" );
 
 	if ( !g->noInclude ) {
 	    /*
--- qt3/src/3rdparty/sqlite/btree_rb.c~	2011-10-30 19:55:57.000000000 +0100
+++ qt3/src/3rdparty/sqlite/btree_rb.c	2013-11-07 23:42:04.724401237 +0100
@@ -311,7 +311,7 @@
 static void print_node(BtRbNode *pNode)
 {
     char * str = append_node(0, pNode, 0);
-    printf(str);
+    printf("%s", str);
 
     /* Suppress a warning message about print_node() being unused */
     (void)print_node;
