--- qt-x11-free-3.0.1/src/tools/qlibrary_unix.cpp.gcc31	Mon Dec 17 15:24:42 2001
+++ qt-x11-free-3.0.1/src/tools/qlibrary_unix.cpp	Mon Dec 17 15:25:25 2001
@@ -104,7 +104,7 @@
 
     QString filename = library->library();
 
-    pHnd = dlopen( filename.latin1() , RTLD_LAZY );
+    pHnd = dlopen( filename.latin1() , RTLD_LAZY|RTLD_GLOBAL );
 #if defined(QT_DEBUG) || defined(QT_DEBUG_COMPONENT)
     if ( !pHnd )
 	qWarning( "%s", dlerror() );
