diff -urN qt-ok/src/tools/qmap.h qt-x11-free-3.3.8b/src/tools/qmap.h
--- qt-ok/src/tools/qmap.h	2008-01-15 20:09:13.000000000 +0100
+++ qt-x11-free-3.3.8b/src/tools/qmap.h	2011-04-13 22:49:13.651871103 +0200
@@ -52,6 +52,7 @@
 #ifndef QT_NO_STL
 #include <iterator>
 #include <map>
+#include <stddef.h>
 #endif
 
 //#define QT_CHECK_MAP_RANGE
diff -urN qt-ok/src/tools/qvaluelist.h qt-x11-free-3.3.8b/src/tools/qvaluelist.h
--- qt-ok/src/tools/qvaluelist.h	2011-04-13 22:45:09.000000000 +0200
+++ qt-x11-free-3.3.8b/src/tools/qvaluelist.h	2011-04-13 22:49:52.386976912 +0200
@@ -50,6 +50,7 @@
 #ifndef QT_NO_STL
 #include <iterator>
 #include <list>
+#include <stddef.h>
 #endif
 
 //#define QT_CHECK_VALUELIST_RANGE
diff -urN qt-ok/src/tools/qvaluevector.h qt-x11-free-3.3.8b/src/tools/qvaluevector.h
--- qt-ok/src/tools/qvaluevector.h	2008-01-15 20:09:13.000000000 +0100
+++ qt-x11-free-3.3.8b/src/tools/qvaluevector.h	2011-04-13 22:49:25.125729465 +0200
@@ -47,6 +47,7 @@
 
 #ifndef QT_NO_STL
 #include <vector>
+#include <stddef.h>
 #endif
 
 template <class T>
