diff -urN qt-x11-free-3.0.2.orig/configure qt-x11-free-3.0.2/configure
--- qt-x11-free-3.0.2.orig/configure	Fri Feb 15 09:12:23 2002
+++ qt-x11-free-3.0.2/configure	Tue Feb 26 23:52:38 2002
@@ -1655,7 +1655,8 @@
         echo >>$mkfile
 	sed -e "s,@REL_QTDIR@,$relpath,g" -e "s,@OUT_QTDIR@,$outpath,g" \
 	    -e "s,@QMAKE_CFLAGS@,$EXTRA_CFLAGS," -e "s,@QMAKE_LFLAGS@,$EXTRA_LFLAGS," \
-	    -e "s,@QT_INSTALL_BINS@,$BIN_DIR,g" -e "s,@QT_INSTALL_PREFIX@,$QT_INSTALL_PREFIX,g" \
+	    -e "s,@QT_INSTALL_BINS@,\$(INSTALL_ROOT)$QMAKE_BIN_DIR,g" \
+	    -e "s,@QT_INSTALL_DATA@,\$(DESTIDR)$QMAKE_DATA_DIR,g" \
 	    -e "s,@QMAKESPEC@,$QMAKESPEC,g" $in_mkfile >>$mkfile
     done
 
diff -urN qt-x11-free-3.0.2.orig/qmake/generators/makefile.cpp qt-x11-free-3.0.2/qmake/generators/makefile.cpp
--- qt-x11-free-3.0.2.orig/qmake/generators/makefile.cpp	Mon Feb 18 13:46:23 2002
+++ qt-x11-free-3.0.2/qmake/generators/makefile.cpp	Tue Feb 26 23:36:53 2002
@@ -1342,7 +1342,7 @@
 	}
 
 	bool do_default = TRUE;
-	QString target, dst=project->variables()[pvar].first();
+	QString target, dst=QString("$(INSTALL_ROOT)") +project->variables()[pvar].first();
 	if(dst.right(1) != Option::dir_sep)
 	    dst += Option::dir_sep;
  	QStringList tmp, &uninst = project->variables()[(*it) + ".uninstall"];
diff -urN qt-x11-free-3.0.2.orig/qmake/generators/unix/unixmake.cpp qt-x11-free-3.0.2/qmake/generators/unix/unixmake.cpp
--- qt-x11-free-3.0.2.orig/qmake/generators/unix/unixmake.cpp	Mon Feb 18 13:46:23 2002
+++ qt-x11-free-3.0.2/qmake/generators/unix/unixmake.cpp	Tue Feb 26 23:37:03 2002
@@ -343,7 +343,9 @@
     QString ret, destdir=project->first("INSTALL_ROOT");
     if(!destdir.isEmpty() && destdir.right(1) != Option::dir_sep)
 	destdir += Option::dir_sep;
-    QString targetdir = Option::fixPathToTargetOS(project->first("target.path"), FALSE);
+    QString targetdir = Option::fixPathToTargetOS(
+        QString("$(INSTALL_ROOT)")
+        + project->first("target.path"), FALSE);
     if(targetdir.right(1) != Option::dir_sep)
 	targetdir += Option::dir_sep;
 
