--- qt-x11-free-3.3.3/src/codecs/qtextcodec.cpp.orig	2004-09-14 11:56:04.232606504 +0200
+++ qt-x11-free-3.3.3/src/codecs/qtextcodec.cpp	2004-09-14 11:57:08.596821656 +0200
@@ -2696,7 +2696,9 @@
 #else
 
 #if defined (_XOPEN_UNIX) && !defined(Q_OS_QNX6) && !defined(Q_OS_OSF)
+    char *oldLocale = setlocale(LC_CTYPE, "");
     char *charset = nl_langinfo (CODESET);
+    setlocale(LC_CTYPE, oldLocale);
     if ( charset )
       localeMapper = QTextCodec::codecForName( charset );
 #endif
