--- qt-copy/src/codecs/qtextcodec.cpp.orig	2006-04-09 12:52:26.000000000 +0200
+++ qt-copy/src/codecs/qtextcodec.cpp	2006-04-09 13:21:27.000000000 +0200
@@ -2784,7 +2784,9 @@
 #else
 
 #if defined (_XOPEN_UNIX) && !defined(Q_OS_QNX6) && !defined(Q_OS_OSF) && !defined(Q_OS_MAC)
+    char *oldLocale = setlocale(LC_CTYPE, "");
     char *charset = nl_langinfo (CODESET);
+    setlocale(LC_CTYPE, oldLocale);
     if ( charset )
         localeMapper = QTextCodec::codecForName( charset );
 #endif
