diff -urNp -x '*.orig' qt3.org/tools/assistant/main.cpp qt3/tools/assistant/main.cpp
--- qt3.org/tools/assistant/main.cpp	2011-10-30 19:55:58.000000000 +0100
+++ qt3/tools/assistant/main.cpp	2021-08-17 10:20:51.821943941 +0200
@@ -289,11 +289,11 @@ int main( int argc, char ** argv )
 	resourceDir = qInstallPathTranslations();
 
     QTranslator translator( 0 );
-    translator.load( QString("assistant_") + QTextCodec::locale(), resourceDir );
+    translator.load( QString("assistant"), resourceDir + QString("/") + QString::fromLatin1(QTextCodec::locale()) + QString("/LC_MESSAGES") );
     a.installTranslator( &translator );
 
     QTranslator qtTranslator( 0 );
-    qtTranslator.load( QString("qt_") + QTextCodec::locale(), resourceDir );
+    qtTranslator.load( QString("qt"), resourceDir + QString("/") + QString::fromLatin1(QTextCodec::locale()) + QString("/LC_MESSAGES") );
     a.installTranslator( &qtTranslator );
 
 
diff -urNp -x '*.orig' qt3.org/tools/designer/app/main.cpp qt3/tools/designer/app/main.cpp
--- qt3.org/tools/designer/app/main.cpp	2011-10-30 19:55:58.000000000 +0100
+++ qt3/tools/designer/app/main.cpp	2021-08-17 10:20:51.821943941 +0200
@@ -48,6 +48,7 @@
 #include <qsettings.h>
 #include <qsplashscreen.h>
 #include <qdir.h>
+#include <qtextcodec.h>
 
 #include <stdlib.h>
 #include <signal.h>
@@ -186,6 +187,14 @@ int main( int argc, char *argv[] )
 	signal( SIGTERM, exitHandler );
     }
 
+    QString resourceDir = qInstallPathTranslations();
+    
+    QTranslator translator( 0 );
+    translator.load( QString("designer"), resourceDir + QString("/") + QString(QTextCodec::locale()) + QString("/LC_MESSAGES") );
+    a.installTranslator( &translator );
+    QTranslator qtTranslator( 0 );
+    qtTranslator.load( QString("qt"), resourceDir + QString("/") + QString(QTextCodec::locale()) + QString("/LC_MESSAGES") );
+    
     extern void qInitImages_designercore();
     qInitImages_designercore();
 
diff -urNp -x '*.orig' qt3.org/tools/linguist/linguist/main.cpp qt3/tools/linguist/linguist/main.cpp
--- qt3.org/tools/linguist/linguist/main.cpp	2011-10-30 19:55:58.000000000 +0100
+++ qt3/tools/linguist/linguist/main.cpp	2021-08-17 10:20:51.821943941 +0200
@@ -41,6 +41,7 @@
 #include <qtranslator.h>
 #include <qsettings.h>
 #include <qsplashscreen.h>
+#include <qtextcodec.h>
 
 int main( int argc, char **argv )
 {
@@ -48,10 +49,16 @@ int main( int argc, char **argv )
     QApplication::setOverrideCursor( Qt::waitCursor );
 
 
+    QString resourceDir = qInstallPathTranslations();
     QTranslator translator( 0 );
-    translator.load( QString( "linguist_" ) + QTextCodec::locale(), "." );
+    translator.load( QString( "linguist" ), resourceDir + QString("/") + QString(QTextCodec::locale()) + QString("/LC_MESSAGES") );
     app.installTranslator( &translator );
 
+    QTranslator qttranslator( 0 );
+    qttranslator.load( QString( "qt" ), resourceDir + QString("/") + QString(QTextCodec::locale()) + QString("/LC_MESSAGES") );
+    app.installTranslator( &qttranslator );
+		
+
     bool showSplash = TRUE;
 
     QString keybase("/Qt Linguist/3.1/");
