--- qt-x11-free-3.2.0/tools/assistant/main.cpp.orig	2003-08-09 13:32:43.000000000 +0200
+++ qt-x11-free-3.2.0/tools/assistant/main.cpp	2003-08-09 13:56:54.000000000 +0200
@@ -249,11 +249,11 @@
 	fprintf( stderr, "Resource file directory '%s' does not exist!\n", resourceDir.latin1() );
 
     QTranslator translator( 0 );
-    translator.load( QString("assistant_") + QTextCodec::locale(), resourceDir );
+    translator.load( QString("assistant"), resourceDir + QString("/") + QString::fromLatin1(QTextCodec::locale()) + QString("/LC_MESSAGES") );
     a.installTranslator( &translator );
 
     QTranslator qtTranslator( 0 );
-    qtTranslator.load( QString("qt_") + QTextCodec::locale(), resourceDir );
+    qtTranslator.load( QString("qt"), resourceDir + QString("/") + QString::fromLatin1(QTextCodec::locale()) + QString("/LC_MESSAGES") );
     a.installTranslator( &qtTranslator );
 
     Config *conf = Config::loadConfig( profileName );
--- qt-x11-free-3.2.0/tools/linguist/linguist/main.cpp.orig	2003-08-09 14:02:15.000000000 +0200
+++ qt-x11-free-3.2.0/tools/linguist/linguist/main.cpp	2003-08-09 14:26:55.000000000 +0200
@@ -42,9 +42,14 @@
 
 
     QTranslator translator( 0 );
-    translator.load( QString( "linguist_" ) + QTextCodec::locale(), "." );
+    translator.load( QString( "linguist" ), resourceDir + QString("/") + QString::fromLatin1(QTextCodec::locale()) + QString("/LC_MESSAGES") );
     app.installTranslator( &translator );
 
+    QTranslator translator( 0 );
+    translator.load( QString( "qt" ), resourceDir + QString("/") + QString::fromLatin1(QTextCodec::locale()) + QString("/LC_MESSAGES") );
+    app.installTranslator( &translator );
+		
+
     bool showSplash = TRUE;
 
     QString keybase("/Qt Linguist/3.1/");
--- qt-x11-free-3.2.0/tools/designer/app/main.cpp.orig	2003-05-21 09:13:12.000000000 +0200
+++ qt-x11-free-3.2.0/tools/designer/app/main.cpp	2003-08-12 09:36:06.090099208 +0200
@@ -179,6 +179,14 @@
 	signal( SIGTERM, exitHandler );
     }
 
+    QString resourceDir = qInstallPathTranslations();
+    
+    QTranslator translator( 0 );
+    translator.load( QString("designer"), resourceDir + QString("/") + QString::fromLatin1(QTextCodec::locale()) + QString("/LC_MESSAGES") );
+    a.installTranslator( &translator );
+    QTranslator qtTranslator( 0 );
+    qtTranslator.load( QString("qt"), resourceDir + QString("/") + QString::fromLatin1(QTextCodec::locale()) + QString("/LC_MESSAGES") );
+    
     extern void qInitImages_designercore();
     qInitImages_designercore();
 
