--- qt-1.43/configs/linux-g++-shared	Mon Feb  1 12:17:17 1999
+++ qt-1.43.new/configs/linux-g++-shared	Fri Mar 12 13:21:30 1999
@@ -59,11 +59,11 @@
 SYSCONF_LINK_LIB_STATIC	= rm -f $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) ; \
 				 $(SYSCONF_AR) $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) $(OBJECTS) $(OBJMOC) 
 # Compiling application source
-SYSCONF_CXXFLAGS	= -pipe -O2 -fno-strength-reduce
+SYSCONF_CXXFLAGS	= -pipe $(RPM_OPT_FLAGS)
 # Compiling library source
-SYSCONF_CXXFLAGS_LIB	= -O2 -fno-strength-reduce -fPIC
+SYSCONF_CXXFLAGS_LIB	= $(RPM_OPT_FLAGS) -fPIC
 # Compiling shared-object source
-SYSCONF_CXXFLAGS_SHOBJ	= -O2 -fno-strength-reduce -fPIC
+SYSCONF_CXXFLAGS_SHOBJ	= $(RPM_OPT_FLAGS) -fPIC
 # Default link type (stati linking is still be used where required)
 SYSCONF_LINK_LIB	= $(SYSCONF_LINK_LIB_SHARED)
 SYSCONF_LINK_TARGET	= $(SYSCONF_LINK_TARGET_SHARED)
