diff -urN qt-x11-free-3.0.2-snapshot-20020222.orig/src/sql/drivers/psql/qsql_psql.cpp qt-x11-free-3.0.2-snapshot-20020222/src/sql/drivers/psql/qsql_psql.cpp
--- qt-x11-free-3.0.2-snapshot-20020222.orig/src/sql/drivers/psql/qsql_psql.cpp	Mon Feb 25 00:09:21 2002
+++ qt-x11-free-3.0.2-snapshot-20020222/src/sql/drivers/psql/qsql_psql.cpp	Mon Feb 25 00:12:16 2002
@@ -40,11 +40,11 @@
 #include <qregexp.h>
 #include <qdatetime.h>
 #include <qpointarray.h>
-#include <postgres.h>
-#include <libpq/libpq-fs.h>
-#include <catalog/pg_type.h>
+#include <postgresql/server/postgres.h>
+#include <postgresql/server/libpq/libpq-fs.h>
+#include <postgresql/server/catalog/pg_type.h>
 #ifndef Q_WS_MAC
-#include <utils/geo_decls.h>
+#include <postgresql/server/utils/geo_decls.h>
 #endif
 #include <math.h>
 
