diff -urN qt-x11-free-3.0.2.orig/src/sql/drivers/psql/qsql_psql.cpp qt-x11-free-3.0.2/src/sql/drivers/psql/qsql_psql.cpp
--- qt-x11-free-3.0.2.orig/src/sql/drivers/psql/qsql_psql.cpp	Mon Jan 21 16:50:44 2002
+++ qt-x11-free-3.0.2/src/sql/drivers/psql/qsql_psql.cpp	Sat Feb 16 15:46:02 2002
@@ -347,6 +347,8 @@
     case QVariant::Invalid:
 #ifdef QT_CHECK_RANGE
 	qWarning("QPSQLResult::data: unknown data type");
+#else
+	;
 #endif
     }
     return QVariant();
