diff -urN qt-x11-free-3.0.2-snapshot-20020222.orig/src/qtmain.pro qt-x11-free-3.0.2-snapshot-20020222/src/qtmain.pro
--- qt-x11-free-3.0.2-snapshot-20020222.orig/src/qtmain.pro	Fri Jan 25 20:24:45 2002
+++ qt-x11-free-3.0.2-snapshot-20020222/src/qtmain.pro	Mon Feb 25 13:45:23 2002
@@ -16,3 +16,8 @@
 	OBJECTS_DIR	= tmp
 }
 win32-borland:INCLUDEPATH += kernel
+
+unix {
+	QMAKE_CFLAGS += -I/usr/include/postgresql/server
+	QMAKE_CXXFLAGS += -I/usr/include/postgresql/server
+}
--- qt-copy/src/qt.pro.orig	Fri Sep  6 14:31:40 2002
+++ qt-copy/src/qt.pro	Sat Sep 28 16:17:36 2002
@@ -91,6 +91,9 @@
 	XML_H		= $$XML_CPP
 	STYLES_H	= $$STYLES_CPP
 	!embedded:!mac:CONFIG	   += x11 x11inc
+	
+	QMAKE_CFLAGS += -I/usr/include/postgresql/server
+	QMAKE_CXXFLAGS += -I/usr/include/postgresql/server	
 }
 
 aix-g++ {
--- qt-copy/plugins/src/sqldrivers/psql/psql.pro.orig       Thu Feb 21 02:11:16 2002
+++ qt-copy/plugins/src/sqldrivers/psql/psql.pro    Mon Feb 25 13:47:12 2002
@@ -8,6 +8,10 @@
 	!contains( LIBS, .*pq.* ) {
 	    LIBS	*= -lpq
 	}
+
+        QMAKE_CFLAGS += -I/usr/include/postgresql/server
+        QMAKE_CXXFLAGS += -I/usr/include/postgresql/server
+
 }
 
 win32 {
