--- qt-copy/src/kernel/qrichtext.cpp.orig	21 Dec 2003 00:48:00 -0000	1.66
+++ qt-copy/src/kernel/qrichtext.cpp	23 Mar 2004 14:47:18 -0000
@@ -1,5 +1,5 @@
 /****************************************************************************
-** 
+**
 **
 ** Implementation of the internal Qt classes dealing with rich text
 **
@@ -6291,8 +6291,16 @@ QString QTextFormat::makeFormatChangeTag
 	    s += QString(!!s?";":"") + "font-size:" + QString::number( fn.pointSize() ) + "pt";
 	if ( font().weight() != defaultFormat->font().weight() )
 	    s += QString(!!s?";":"") + "font-weight:" + QString::number( fn.weight() * 8 );
+	QString d;
+	bool none = false;
 	if ( font().underline() != defaultFormat->font().underline() )
-	    s += QString(!!s?";":"") + "text-decoration:" + ( font().underline() ? "underline" : "none");
+	    font().underline() ? d += "underline" : none = true;
+	if ( font().strikeOut() != defaultFormat->font().strikeOut() )
+	    font().strikeOut() ? d += QString(!!d?" ":"") + "line-through" : none = true;
+	if ( !d.isNull() )
+	    s += QString(!!s?";":"") + "text-decoration:" + d;
+	else if ( none )
+	    s += QString(!!s?";":"") + "text-decoration:none";
 	if ( vAlign() != defaultFormat->vAlign() ) {
 	    s += QString(!!s?";":"") + "vertical-align:";
 	    if ( vAlign() == QTextFormat::AlignSuperScript )
@@ -6418,7 +6426,7 @@ QTextFormat QTextFormat::makeTextFormat(
 		format.logicalFontSize = 0;
 		int size = int( scaleFontsFactor * style.mid( 10, style.length() - 12 ).toDouble() );
 		format.setPointSize( size );
-	    } if ( style.startsWith("font-style:" ) ) {
+	    } else if ( style.startsWith("font-style:" ) ) {
 		QString s = style.mid( 11 ).stripWhiteSpace();
 		if ( s == "normal" )
 		    format.fn.setItalic( FALSE );
@@ -6438,7 +6446,8 @@ QTextFormat QTextFormat::makeTextFormat(
 		format.fn.setFamily( family );
 	    } else if ( style.startsWith("text-decoration:" ) ) {
 		QString s = style.mid( 16 ).stripWhiteSpace();
-		format.fn.setUnderline( s == "underline" );
+		format.fn.setUnderline( s.find( "underline" ) > -1 );
+		format.fn.setStrikeOut( s.find( "line-through" ) > -1 );
 	    } else if ( style.startsWith("vertical-align:" ) ) {
 		QString s = style.mid( 15 ).stripWhiteSpace();
 		if ( s == "sub" )
