--- qt-copy/src/kernel/qfont_x11.cpp.orig	Thu Oct 24 19:13:26 2002
+++ qt-copy/src/kernel/qfont_x11.cpp	Thu Oct 24 20:31:54 2002
@@ -2404,9 +2404,9 @@
     case QFont::CanadianAboriginal:        ch = 0x1410; break;
     case QFont::Mongolian:                 ch = 0x1800; break;
 
-    case QFont::CurrencySymbols:           ch = 0x20aa; break;
-    case QFont::LetterlikeSymbols:         ch = 0x2122; break;
-    case QFont::NumberForms:               ch = 0x215b; break;
+    case QFont::CurrencySymbols:           ch = 0x20a0; break;
+    case QFont::LetterlikeSymbols:         ch = 0x2100; break;
+    case QFont::NumberForms:               ch = 0x2160; break;
     case QFont::MathematicalOperators:     ch = 0x222b; break;
     case QFont::TechnicalSymbols:          ch = 0x2440; break;
     case QFont::GeometricSymbols:          ch = 0x2500; break;
