diff -urNp -x '*.orig' qt3.org/configure qt3/configure
--- qt3.org/configure	2011-10-30 19:55:56.000000000 +0100
+++ qt3/configure	2021-08-17 10:20:52.835277924 +0200
@@ -2665,6 +2665,9 @@ if true; then ###[ '!' -f "$outpath/bin/
 	    EXTRA_CFLAGS="$EXTRA_CFLAGS \$(QMAKE_CFLAGS_RELEASE)"
 	    EXTRA_CXXFLAGS="$EXTRA_CXXFLAGS \$(QMAKE_CXXFLAGS_RELEASE)"
         fi
+	if [ -n "$OPTFLAGS" ]; then
+	    EXTRA_CFLAGS="$EXTRA_CFLAGS $OPTFLAGS"
+	fi
 	if [ "$PLATFORM_MAC" = "yes" ]; then
 	    echo "CARBON_LFLAGS +=-framework CoreServices" >>$mkfile
 	    EXTRA_LFLAGS="$EXTRA_LFLAGS \$(CARBON_LFLAGS)"
