--- qt-copy/extensions/nsplugin/src/qnp.pro~	2003-03-03 17:44:27.000000000 +0100
+++ qt-copy/extensions/nsplugin/src/qnp.pro	2003-04-17 17:47:16.000000000 +0200
@@ -1,6 +1,5 @@
 TEMPLATE    	= lib
-CONFIG -= dll 
-CONFIG      	+= qt x11 release staticlib
+CONFIG      	+= qt x11 release
 SOURCES		= qnp.cpp
 unix:HEADERS += qnp.h
 win32:HEADERS	= ../../../include/qnp.h
--- qt-x11-free-3.1.1/tools/assistant/lib/lib.pro~	Thu Sep  5 13:56:11 2002
+++ qt-x11-free-3.1.1/tools/assistant/lib/lib.pro	Wed Mar  5 14:28:50 2003
@@ -5,9 +5,6 @@
 SOURCES		= qassistantclient.cpp
 HEADERS         += $$QT_SOURCE_TREE/include/qassistantclient.h
 
-CONFIG += staticlib
-CONFIG	-= dll
-
 DEFINES		+= QT_INTERNAL_NETWORK
 include( ../../../src/qt_professional.pri )
 
--- qt-copy/tools/designer/designer/designer.pro~	2003-05-17 01:05:40.000000000 +0200
+++ qt-copy/tools/designer/designer/designer.pro	2003-05-17 01:12:28.000000000 +0200
@@ -1,6 +1,5 @@
 TEMPLATE	= lib
-CONFIG		+= qt warn_on staticlib qmake_cache
-CONFIG 		-= dll
+CONFIG		+= qt warn_on qmake_cache
 !force_static:!win32:contains(QT_PRODUCT,qt-internal) {
    CONFIG          -= staticlib
    CONFIG          += dll 
@@ -426,7 +425,7 @@
 		LIBS	+= $$QT_BUILD_TREE/lib/libqui.a $$QT_BUILD_TREE/lib/libqassistantclient.a
       } else :LIBS	+= $$QT_BUILD_TREE/lib/qui.lib $$QT_BUILD_TREE/lib/qassistantclient.lib 
 }
-unix:LIBS		+= -L$$QT_BUILD_TREE/lib -lqui $$QT_BUILD_TREE/lib/libqassistantclient.a
+unix:LIBS		+= -L$$QT_BUILD_TREE/lib -lqui $$QT_BUILD_TREE/lib/libqassistantclient.so.1.0.0
 
 TRANSLATIONS	= designer_de.ts designer_fr.ts
 
--- qt-x11-free-3.1.1/tools/designer/editor/editor.pro~	Tue Nov 19 22:07:43 2002
+++ qt-x11-free-3.1.1/tools/designer/editor/editor.pro	Wed Mar  5 14:23:47 2003
@@ -1,6 +1,5 @@
 TEMPLATE	= lib
-CONFIG		+= qt warn_on staticlib
-CONFIG		-= dll
+CONFIG		+= qt warn_on
 HEADERS		= editor.h  \
 		  parenmatcher.h  \
 		  completion.h \
--- qt-x11-free-3.1.2/tools/designer/plugins/cppeditor/cppeditor.pro~	Tue Jan 14 18:56:20 2003
+++ qt-x11-free-3.1.2/tools/designer/plugins/cppeditor/cppeditor.pro	Fri Mar  7 07:58:38 2003
@@ -13,7 +13,7 @@
 win32{
     LIBS	+= $$QT_BUILD_TREE/lib/editor.lib
 } else {
-    LIBS	+= -L$$QT_BUILD_TREE/lib $$QT_BUILD_TREE/lib/libeditor.a
+    LIBS	+= -L$$QT_BUILD_TREE/lib $$QT_BUILD_TREE/lib/libeditor.so.1.0.0
 }
 
 target.path += $$plugins.path/designer
