diff -urN qt-x11-free-3.0.1.org/mkspecs/linux-g++/qmake.conf qt-x11-free-3.0.1/mkspecs/linux-g++/qmake.conf
--- qt-x11-free-3.0.1.org/mkspecs/linux-g++/qmake.conf	Mon Jan 28 00:35:43 2002
+++ qt-x11-free-3.0.1/mkspecs/linux-g++/qmake.conf	Mon Jan 28 00:36:41 2002
@@ -44,8 +44,8 @@
 QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
 QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
 
-QMAKE_LINK		= g++
-QMAKE_LINK_SHLIB	= g++
+QMAKE_LINK		= objprelink $(OBJECTS) $(OBJMOC) && g++
+QMAKE_LINK_SHLIB	= objprelink $(OBJECTS) $(OBJMOC) && g++
 QMAKE_LFLAGS		= 
 QMAKE_LFLAGS_RELEASE	=
 QMAKE_LFLAGS_DEBUG	= 
