--- qt-copy/configure.orig	2006-04-09 13:23:21.000000000 +0200
+++ qt-copy/configure	2006-04-09 13:26:51.000000000 +0200
@@ -2641,7 +2641,7 @@
 	[ -z "$QMAKE_DATA_DIR" ] && QMAKE_DATA_DIR="${QT_INSTALL_PREFIX}"
         echo >>$mkfile
 	sed -e "s,@SOURCE_PATH@,$relpath,g" -e "s,@BUILD_PATH@,$outpath,g" \
-	    -e "s,@QMAKE_CFLAGS@,$EXTRA_CFLAGS,g" -e "s,@QMAKE_LFLAGS@,$EXTRA_LFLAGS,g" \
+	    -e "s:@QMAKE_CFLAGS@:$EXTRA_CFLAGS:g" -e "s,@QMAKE_LFLAGS@,$EXTRA_LFLAGS,g" \
             -e "s,@QMAKE_CXXFLAGS@,$EXTRA_CXXFLAGS,g" \
 	    -e "s,@QT_INSTALL_BINS@,\$(INSTALL_ROOT)$QMAKE_BIN_DIR,g" \
 	    -e "s,@QT_INSTALL_DATA@,\$(INSTALL_ROOT)$QMAKE_DATA_DIR,g" \
