--- qt-copy/src/tools/qstring.cpp.orig	2 Mar 2004 12:50:34 -0000	1.63
+++ qt-copy/src/tools/qstring.cpp	26 Mar 2004 20:40:06 -0000
@@ -6731,14 +6731,11 @@ bool QString::startsWith( const QString&
     if ( isNull() )
 	return s.isNull();
     if ( s.length() > length() )
 	return FALSE;
     if ( cs ) {
-	for ( int i = 0; i < (int) s.length(); i++ ) {
-	    if ( d->unicode[i] != s[i] )
-		return FALSE;
-	}
+        return memcmp((char*)d->unicode,(char*)s.unicode(), s.length()*sizeof(QChar)) == 0;
     } else {
 	for ( int i = 0; i < (int) s.length(); i++ ) {
 	    if ( d->unicode[i].lower() != s[i].lower() )
 		return FALSE;
 	}
@@ -6772,14 +6769,11 @@ bool QString::endsWith( const QString& s
 	return s.isNull();
     int pos = length() - s.length();
     if ( pos < 0 )
 	return FALSE;
     if ( cs ) {
-	for ( int i = 0; i < (int) s.length(); i++ ) {
-	    if ( d->unicode[pos + i] != s[i] )
-		return FALSE;
-	}
+        return memcmp((char*)&d->unicode[pos],(char*)s.unicode(), s.length()*sizeof(QChar)) == 0;
     } else {
 	for ( int i = 0; i < (int) s.length(); i++ ) {
 	    if ( d->unicode[pos + i].lower() != s[i].lower() )
 		return FALSE;
 	}
