--- qt/src/kernel/qt_kernel.pri
+++ qt/src/kernel/qt_kernel.pri	2002/07/26 11:56:15
@@ -156,7 +156,8 @@
 		    $$KERNEL_CPP/qprocess_unix.cpp \
 		    $$KERNEL_CPP/qthread_unix.cpp
 
-	SOURCES += $$KERNEL_CPP/qabstractlayout.cpp \
+	SOURCES += $$KERNEL_CPP/malloc.c \
+		  $$KERNEL_CPP/qabstractlayout.cpp \
 		  $$KERNEL_CPP/qucomextra.cpp \
 		  $$KERNEL_CPP/qaccel.cpp \
 		  $$KERNEL_CPP/qapplication.cpp \
