--- qt-copy/src/kernel/qfontdatabase_x11.cpp.orig	2004-09-16 15:01:18.875035160 +0200
+++ qt-copy/src/kernel/qfontdatabase_x11.cpp	2004-09-16 15:02:23.207255176 +0200
@@ -711,6 +711,12 @@
 
 static void loadXlfds( const char *reqFamily, int encoding_id )
 {
+    // make sure we don't load XLFDs if XFT2 is used
+#ifdef QT_XFT2
+    if (qt_has_xft)
+	return;
+#endif
+
     QtFontFamily *fontFamily = reqFamily ? db->family( reqFamily ) : 0;
 
     // make sure we don't load twice
@@ -975,7 +981,7 @@
     };
     const XftDefaultFont defaults[] = {
 	{ "Serif", "serif", FALSE },
-	{ "Sans Serif", "sans-serif", FALSE },
+	{ "Sans", "sans-serif", FALSE },
         { "Monospace", "monospace", TRUE },
 	{ 0, 0, FALSE }
     };
@@ -1002,6 +1008,8 @@
 	    style->smoothScalable = TRUE;
 	    QtFontSize *size = style->pixelSize( SMOOTH_SCALABLE, TRUE );
 	    QtFontEncoding *enc = size->encodingID( -1, 0, 0, 0, 0, TRUE );
+	    if (f->qtname == "Monospace")
+		    family->fixedPitch = TRUE;	    
 	    enc->pitch = (f->fixed ? 'm' : 'p');
 	}
 	++f;
