--- qt-x11-free-3.3.1/src/kernel/qfontdatabase_x11.cpp.fontdatabase	2004-02-18 14:22:49.000000000 +0100
+++ qt-x11-free-3.3.1/src/kernel/qfontdatabase_x11.cpp	2004-03-05 00:10:57.126934173 +0100
@@ -711,6 +711,12 @@
 
 static void loadXlfds( const char *reqFamily, int encoding_id )
 {
+    // make sure we don't load XLFDs if XFT2 is used
+#ifdef QT_XFT2
+    if (qt_has_xft)
+	return;
+#endif
+
     QtFontFamily *fontFamily = reqFamily ? db->family( reqFamily ) : 0;
 
     // make sure we don't load twice
@@ -943,7 +949,7 @@
     };
     const XftDefaultFont defaults[] = {
 	{ "Serif", "serif", FALSE },
-	{ "Sans Serif", "sans-serif", FALSE },
+	{ "Sans", "sans-serif", FALSE },
         { "Monospace", "monospace", TRUE },
 	{ 0, 0, FALSE }
     };
@@ -967,6 +973,8 @@
 	    style->smoothScalable = TRUE;
 	    QtFontSize *size = style->pixelSize( SMOOTH_SCALABLE, TRUE );
 	    QtFontEncoding *enc = size->encodingID( -1, 0, 0, 0, 0, TRUE );
+	    if (f->qtname == "Monospace")
+		family->fixedPitch = TRUE;
 	    enc->pitch = ( spacing_value >= XFT_CHARCELL ? 'c' :
 			   ( spacing_value >= XFT_MONO ? 'm' : 'p' ) );
 	}
