--- qt-copy-040304/config.tests/x11/xcursor.test.orig	2004-02-05 12:20:34.000000000 +0100
+++ qt-copy-040304/config.tests/x11/xcursor.test	2004-03-04 15:28:48.151859208 +0100
@@ -72,10 +72,10 @@
         XCURSOR=no
 	[ "$VERBOSE" = "yes" ] && \
 	    echo "  Could not find Xcursor version."
-    elif [ "$XCURSOR_MAJOR" != "1" -o "$XCURSOR_MINOR" != "0" ]; then
+    elif [ "$XCURSOR_MAJOR" != "1" -o "$XCURSOR_MINOR" != "0" ] && [ "$XCURSOR_MAJOR" != "1" -o "$XCURSOR_MINOR" != "1" ]; then
         XCURSOR=no
 	[ "$VERBOSE" = "yes" ] && \
-	    echo "  Found Xcursor version $XCURSOR_MAJOR.$XCURSOR_MINOR, version 1.0 required."
+	    echo "  Found Xcursor version $XCURSOR_MAJOR.$XCURSOR_MINOR, version 1.0 or 1.1 required."
     fi
 fi
 
