diff -Nru qt-1.40/Makefile qt-1.40.new/Makefile
--- qt-1.40/Makefile	Thu Jul  9 21:25:06 1998
+++ qt-1.40.new/Makefile	Mon Aug  3 21:42:59 1998
@@ -4,7 +4,7 @@
 # Read PORTING for instructions how to port Qt to a new platform.
 
 
-all: moc src tutorial examples
+all: moc src
 	@echo
 	@echo "The Qt library is now built in ./lib"
 	@echo "The Qt examples are built in the directories in ./examples"
